pkg/query/storeset.go

```diff
@@ -22,6 +22,7 @@ import (
 const (
 	unhealthyStoreMessage = "removing store because it's unhealthy or does not exist"
 	droppingStoreMessage  = "dropping store, external labels are not unique"
+	storeUnhealthyTimeout = 5 * time.Minute
```

Comment:
could you make a flag for it? My all stores are from FILE SD and it's important  for me to view them in UI if they fail..

Reply:
`--store.unhealthy-timeout=5m` there you go !